### PR TITLE
drop-ins: move systemd drop-ins into unified schedule struct

### PR DESCRIPTION
**Pull request by jkellerer**
provide better granularity where drop-in files can be defined (in relation to #217) 

```yaml
global:
  schedule-defaults:
    systemd-drop-in-files:
      - global-default-drop-in.conf
    log: "scheduling.log"
    lock-wait: 120m

__base:
  repository: "file:/mnt/backup"
  systemd-drop-in-files:
    - default-drop-in.conf

default:
  inherit: "__base"

  check:
    schedule:
      at: "monthly"
      systemd-drop-in-files:
        - check-drop-in.conf
```

### CODE REVIEW

File: config/global.go [hunk 1]
```diff
@@ -28,7 +28,6 @@ type Global struct {
 	LegacyArguments      bool                `mapstructure:"legacy-arguments" default:"false" deprecated:"0.20.0" description:"Legacy, broken arguments mode of resticprofile before version 0.15"`
 	SystemdUnitTemplate  string              `mapstructure:"systemd-unit-template" default:"" description:"File containing the go template to generate a systemd unit - see https://creativeprojects.github.io/resticprofile/schedules/systemd/"`
 	SystemdTimerTemplate string              `mapstructure:"systemd-timer-template" default:"" description:"File containing the go template to generate a systemd timer - see https://creativeprojects.github.io/resticprofile/schedules/systemd/"`
-	SystemdDropInFiles   []string            `mapstructure:"systemd-drop-in-files" default:"" description:"Files containing systemd drop-in (override) files - see https://creativeprojects.github.io/resticprofile/schedules/systemd/"`
```
Comment (creativeprojects) [inline config/global.go:31]: `systemd-drop-in` was introduced in `v0.25.0` meaning it's a breaking change if we remove it from here. I know it's a bit annoying 🤔

Comment (creativeprojects) [reply config/global.go:31]: doh! it's now coming from `ScheduleBaseConfig`. Just ignore me, too much booze yesterday 😆

Comment (jkellerer) [reply config/global.go:31]: Guess you've also seen that it came from here: 
https://github.com/creativeprojects/resticprofile/pull/333/files#diff-7332be621f45d9aeebdbcc04e15b4ca8ae863ff8d9a57822405d6c0dfc04f632R31

Therefore it is no issue as it wasn't released yet 🙂. `0.25` brought the profile variant which still exists.

Comment (creativeprojects) [reply config/global.go:31]: sorry yes, I got interrupted yesterday, I thought I deleted the comment 🤷🏻

**Review by creativeprojects — APPROVED** [review]
LGTM!

Thanks 😉

**Pull request merged**